device/platforms/stm32/GPIODevice.h

```diff
@@ -0,0 +1,100 @@
+/**
+ * HAL Implementation for GPIO Devices
+ *
+ * @author Aaron Chan
+ */
+
+#ifndef LAUNCH_CORE_GPIODEVICE_H
```

Comment:
probably want to name this preprocessor define something unique to stm32 platform